### ISSUE TITLE
Allow users to filter files by circles

### ISSUE THREAD

**Issue by dtygel** [nextcloud/circles#137]
Today, the file list view allows users to filter files by tags, but there is no way to list files per circle(s).

We are working on a PR to do this. For this to be possible, we will also need to make a PR to NextCloud Core, including the needed infrastructure in `core/js/files/client.js` and in `apps/dav/lib/Connector/Sabre/FilesReportPlugin`.

Our solution involves the creation of a new Method in Circles API V1, which we called `getObjectIdsForCircles`, and uses another new method, in ShareByCirclesProvider, which retrieves  the shares by Circles's Unique Ids.

**Comment by dtygel** [comment]
@viniciuscb

**Comment by daita** [comment]
About the Circles API, I know it is missing a function to retrieve the list of the shared items (circles_shares), I think nothing read the information listed in that database right now, and I will work on this.

However this will not be necessary in here.

If you start working on the files from Nextcloud, maybe adding a entry on the left menu:

![selection_083](https://user-images.githubusercontent.com/1038617/31347189-a481e10a-ad02-11e7-9506-147111fb029d.png)

A 'shared by ...' entry that allow your users to filter the file shared by another user, a group or a circle. 
This way you use directly the ShareProvider as the shares for the files are in oc_shares and not oc_circles_shares

This would be an interesting feature.

**Comment by daita** [comment]
(Are we talking about the same thing in fact ? ;p)

**Comment by daita** [comment]
@dtygel I still like the idea, any update ? :)

**Comment by dtygel** [comment]
@daita Yes! We have a working code, but haven't done the tests yet. If you'd like to see how it's going, we can make a PR, still without the tests. What do you think?

**Comment by daita** [comment]
Sure, please do. This would be great for NC13

**Comment by dtygel** [comment]
Above are the 3 commits that make the files' panel work. The first one is in _core_. The second and third ones are in circles' plugin.

We'll do both PRs this week or on Monday for your review.

**Comment by daita** [comment]
Wow, this is nice.

I am cleaning few stuff and I will PR on your fork, but can you please rebase on your side ?

**Comment by daita** [comment]
![selection_092](https://user-images.githubusercontent.com/1038617/31818069-b09c43c6-b57d-11e7-8310-c7508969dec5.png)

**Comment by dtygel** [comment]
Cool that you liked it, @daita . We'll then rebase now to develop so that you can create the PR and do the cleanings. Our branch was based on stable12, so we'll create another one.

**Comment by daita** [comment]
I don't think I am able to PR on your fork, this is by work about this:

https://github.com/nextcloud/circles/tree/edit-files-panel

- I don't want to fill SharebyCirclesProvider, which is a provider for OCP\Share20. Also, I want to rewrite this class to use Db/CircleProviderRequest. So I moved everything there.
- I renamed to getFilesFromCircles, even the API so you will need to adjust on nextcloud/server
- Moved templates/css/js to a subfolder.

**Comment by daita** [comment]
https://github.com/coletivoEITA/circles/pull/2

**Comment by viniciuscb** [comment]
We tested, everything seems fine. Great job!! Should we do a PR of this to circles?

Um abraço!

**Comment by daita** [comment]
:+1: 

please, make the PR for circles and nextcloud

**Comment by daita** [comment]
@viniciuscb can you check if the integration of this feature is still working on your side ? it seems to be buggy on NC13 and NC14 !

Obrigado

**Comment by joshtrichards** [comment]
Added in #141 and evolved in subsequent PRs.
